File: bubble_sort/java/bubble_sort.java

import java.util.Arrays;

class BubbleSort {

    // An optimized version of Bubble Sort 
    static void bubbleSort(int arr[]) {
        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < i; j++) {
                if (arr[i] < arr[j]) {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }
    }

    // Function to print an array  
    static void printArray(int arr[]) {
        System.out.println(Arrays.toString(arr));
    }

    // Driver program 
    public static void main(String args[]) {
        int arr[] = {64, 34, 25, 12, 22, 11, 90};
        int n = arr.length;
        bubbleSort(arr);
        System.out.println("Sorted array: ");
        printArray(arr);
    }
}

File: bubble_sort/cpp/bubble_sort.cpp
#include <iostream>



void bubblesort (int n, int a[]) {
    int i, j, t;
    for(i = 0; i < n-1; ++i) {
        for(j = 0; j < n-i-1; ++j) {
            if(a[j+1] < a[j]) {
                t = a[j];
                a[j] = a[j+1];
                a[j+1] = t;
            }
        }
    }
}

void print_array(int a_size, int *a) {
	for(int i=0; i<a_size; i++) {
		std::cout << a[i] << " ";
	}
}

int main() {

	int size = 10;
	int arr[10] = {4, 2, 7, 3, 0, 9, 1, 5, 6, 8};
	bubblesort(size, arr);
	print_array(size, arr);

	return 0;
}
